### PR TITLE
(documentation) Add a "troubleshooting" section to typescript.md

### PR DESCRIPTION
**Pull request by Offirmo**
Hi, I had trouble setting up TypeScript support and had to search a lot to make it work. (credits to https://github.com/storybookjs/storybook/issues/14118)

This should help the next one!

Issue:

Not able to have TypeScript checks

## What I did

Added documentation

## How to test

Not needed

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

**Comment by jonniebigodes** [comment]
@Offirmo sorry for not addressing this sooner. I'll take a look at it tomorrow and work with you on any proposed changes.

Stay safe

File: docs/configure/typescript.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: 'TypeScript'
 ---
 
-Storybook has built-in Typescript support, so your Typescript project should work with zero configuration needed.
+Storybook has built-in Typescript support, so your Typescript project should work with zero configuration needed. The config files from `.storybook` can be in TypeScript as well.
```
Comment (jonniebigodes) [inline docs/configure/typescript.md:5]: @Offirmo if you could revert back to what's currently documented we'd appreciate it. The reasoning behind this is that in the first sentence we're mentioning that Storybook as a whole has Typescript support, despite the config files being created with `.js` extension.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@Offirmo left a couple of items that need a bit of polish, let me know once you address them and we'll go from there.

Stay safe

**Comment by jonniebigodes** [comment]
@Offirmo just a friendly reminder if you're willing to work on this pull request. So that the changes you're introducing will help not only you but the rest of the community.

Let me know and we'll go from there. Sounds good?

Stay safe

**Pull request closed without merging**